curp/src/server/curp_node.rs

```diff
@@ -312,30 +340,58 @@ impl<C: 'static + Command, RC: RoleChange + 'static> CurpNode<C, RC> {
         let mut ticker = tokio::time::interval(heartbeat_interval);
         ticker.set_missed_tick_behavior(MissedTickBehavior::Delay);
         loop {
-            let _now = ticker.tick().await;
+            tokio::select! {
+                _now = ticker.tick() => {}
+                _ = shutdown_listener.wait_self_shutdown() => {
+                    debug!("election task exits");
+                    return;
+                }
+            }
             if let Some(vote) = curp.tick_election() {
                 Self::bcast_vote(curp.as_ref(), &connects, vote).await;
             }
         }
     }
 
     /// Responsible for bringing up `sync_follower_task` when self becomes leader
-    async fn sync_follower_daemon(
+    #[allow(clippy::integer_arithmetic)]
+    async fn sync_followers_daemon(
         curp: Arc<RawCurp<C, RC>>,
-        connect: Arc<impl ConnectApi + ?Sized>,
-        sync_event: Arc<Event>,
+        connects: HashMap<u64, Arc<impl ConnectApi + ?Sized>>,
+        shutdown_trigger: shutdown::Trigger,
     ) {
+        let mut shutdown_listener = shutdown_trigger.subscribe();
         let leader_event = curp.leader_event();
         loop {
             if !curp.is_leader() {
-                leader_event.listen().await;
+                shutdown_trigger.mark_sync_daemon_shutdown();
+                shutdown_trigger.check_and_shutdown();
+                tokio::select! {
+                    _ = shutdown_listener.wait_self_shutdown() => {
+                        debug!("sync follower daemon exits");
+                        return;
+                    }
+                    _ = leader_event.listen() => {
+                        shutdown_trigger.mark_sync_daemon_shutdown();
```

Comment:
This should be `reset_sync_daemon_shutdown`

Reply:
Fixed